libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/rest/RestClientTest.java

```diff
@@ -552,11 +556,166 @@ public void testFileUpload() throws Exception {
         assertFalse("File should not exist", file.exists());
     }
 
+    /**
+     * Tests if a stream from {@link RestResponse#asInputStream()} is readable.
+     *
+     * @throws Exception
+     */
+    public void testResponseStreamIsReadable() throws Exception {
+        final RestResponse response = getStreamTestResponse();
+
+        try (InputStream in = response.asInputStream()) {
```

Comment:
@clibc I don't know if we enforce ```Java 7``` as a minimum, so it might be best to go with the ```Java 6``` syntax here.